src/miner.cpp

```diff
@@ -251,9 +251,10 @@ CBlockTemplate* Mining::CreateNewBlock(const CChainParams& chainparams) const
             if (!IsFinalTx(tx, nHeight, nLockTimeCutoff))
                 continue;
 
+            const uint64_t maxSigOps = Policy::blockSigOpAcceptLimit(nBlockSize + nTxSize);
```

Comment:
CreateNewBlock is indeed huge. It is also a loop which makes it rather difficult to refactor in a more readable way.  I'm not arguing against a refactor, though ;)